src/db/mod.rs

```diff
@@ -368,23 +368,29 @@ pub mod models;
 
 /// Creates a back-up of the sqlite database
 /// MySQL/MariaDB and PostgreSQL are not supported.
-pub async fn backup_database(conn: &mut DbConn) -> Result<(), Error> {
+pub async fn backup_database(conn: &mut DbConn) -> Result<String, Error> {
     db_run! {@raw conn:
         postgresql, mysql {
             let _ = conn;
             err!("PostgreSQL and MySQL/MariaDB do not support this backup feature");
         }
         sqlite {
-            use std::path::Path;
-            let db_url = CONFIG.database_url();
-            let db_path = Path::new(&db_url).parent().unwrap().to_string_lossy();
-            let file_date = chrono::Utc::now().format("%Y%m%d_%H%M%S").to_string();
-            diesel::sql_query(format!("VACUUM INTO '{db_path}/db_{file_date}.sqlite3'")).execute(conn)?;
-            Ok(())
+            backup_sqlite_database(conn)
         }
     }
 }
 
+#[cfg(sqlite)]
+pub fn backup_sqlite_database(conn: &mut diesel::sqlite::SqliteConnection) -> Result<String, Error> {
+    use diesel::RunQueryDsl;
+    let db_url = CONFIG.database_url();
+    let db_path = std::path::Path::new(&db_url).parent().unwrap().to_string_lossy();
+    let file_date = chrono::Utc::now().format("%Y%m%d_%H%M%S").to_string();
+    let backup_file = format!("{db_path}/db_{file_date}.sqlite3");
```

Comment:
I think we should be using path.join() here instead of format!() to add the path separator, something like this (haven't tested it)
```suggestion
    let db_path = std::path::Path::new(&db_url).parent().unwrap();
    let file_date = chrono::Utc::now().format("%Y%m%d_%H%M%S");
    let backup_file = db_path.join(format!("db_{file_date}.sqlite3"));
```

Reply:
Ah, yea, that might be a more safe option indeed.
Ill go and test it.